lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,327 @@
+  late ScrollController scrollController;
+  final Set<int> selectedUserIds = {};
+  List<User> filteredUsers = [];
+
+  @override
+  void initState() {
+    super.initState();
+    searchController = TextEditingController()..addListener(_handleSearchUpdate);
+    scrollController = ScrollController();
+  }
+
+  @override
+  void dispose() {
+    searchController.dispose();
+    scrollController.dispose();
+    super.dispose();
+  }
+
+  @override
+  void onNewStore() {
+    final store = PerAccountStoreWidget.of(context);
+    _updateFilteredUsers(store);
+  }
+
+  void _handleSearchUpdate() {
+    final store = PerAccountStoreWidget.of(context);
+    _updateFilteredUsers(store);
+  }
+
+  // function to sort users based on recency of DM's
+  void _updateFilteredUsers(PerAccountStore store) {
+    setState(() {
+      filteredUsers = store.users.values
+        .where((user) =>
+          user.userId != store.selfUserId &&
+          user.fullName.toLowerCase().contains(searchController.text.toLowerCase()))
+        .toList()
+        ..sort((a, b) {
+          final recentDms = store.recentDmConversationsView;
+          final aLatestMessageId = recentDms.latestMessagesByRecipient[a.userId];
+          final bLatestMessageId = recentDms.latestMessagesByRecipient[b.userId];
+
+          if (aLatestMessageId != null && bLatestMessageId != null) {
+            return bLatestMessageId.compareTo(aLatestMessageId);
+          }
+          if (aLatestMessageId != null) return -1;
+          if (bLatestMessageId != null) return 1;
+          return 0;
+        });
+    });
+  }
+
+  // Scroll to the search field when the user selects a user
+  void scrollToSearchField() {
+    SchedulerBinding.instance.addPostFrameCallback((_) {
+      if (scrollController.hasClients) {
+        scrollController.animateTo(
+          scrollController.position.maxScrollExtent,
+          duration: const Duration(milliseconds: 200),
+          curve: Curves.easeOut);
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      height: MediaQuery.of(context).size.height * 0.95,
+      child: Column(children: [
+        NewDmHeader(
+          selectedUserIds: selectedUserIds,
+          onClose: () => Navigator.pop(context)),
+        NewDmSearchBar(
+          searchController: searchController,
+          scrollController: scrollController,
+          selectedUserIds: selectedUserIds),
+        Expanded(child: NewDmUserList(
+          filteredUsers: filteredUsers,
+          selectedUserIds: selectedUserIds,
+          onUserSelected: (userId) {
+            setState(() {
+              if (selectedUserIds.contains(userId)) {
+                selectedUserIds.remove(userId);
+              } else {
+                selectedUserIds.add(userId);
+              }
+              scrollToSearchField();
+            });
+          }))]));
+  }
+}
+
+class NewDmHeader extends StatelessWidget {
```

Comment:
For helper widgets like this, let's make them private (`_NewDmHeader`) unless it is otherwise necessary.